experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
-                Enough Beams in beamdir"
-            error_count=error_count+1
-
-    # check scan boundary not less than minimum required scan time. 
-    if cpdict['scanboundf']==1: # REVIEW #26 We assume this is a flag, but naming could be better, also can use True and False instead of 1's and 0's if it's a boolean
-        if cpdict['scanbound'] < (len(cpdict['scan'])*cpdict['intt']): # REVIEW #5 Need units documented somewhere highly visible for scanbound, intt
-            error_dict[error_count]="CP Scan Too Long for ScanBoundary"
-            error_count=error_count+1
-
-    #TODO other checks
-
-    #freq=12300 # in kHz
-    #clrfrqf=1 # flag for clear frequency search
-    #clrfrqrange=300 # clear frequency range if searching
-    # receiving params
-    #xcf=1 # flag for cross-correlation
-    #acfint=1 # flag for getting lag-zero power of interferometer
-    #wavetype='SINE'
-    #seqtimer=0
-    # cpid is a unique value? # REVIEW Talked about placing a file with all known cpids in git repo
-
-    return error_dict
+interface_types = frozenset(['SCAN', 'INTTIME', 'INTEGRATION', 'PULSE'])
+
+"""
+INTERFACING TYPES:
+
+NONE : Only the default, must be changed.
+SCAN : Scan by scan interfacing. exp_slice #1 will scan first
+    followed by exp_slice #2 and subsequent exp_slice's.
+INTTIME : nave by nave interfacing (full integration time of
+     one pulse_sequence, then the next). Time/number of pulse_sequences
+    dependent on intt and intn in exp_slice. Effectively
+    simultaneous scan interfacing, interleaving each
+    integration time in the scans. exp_slice #1 first inttime or
+    beam direction will run followed by exp_slice #2's first inttime,
+    etc. if exp_slice #1's len(scan) is greater than exp_slice #2's, exp_slice
+    #2's last integration will run and then all the rest of exp_slice
+    #1's will continue until the full scan is over. exp_slice 1
+    and 2 must have the same scan boundary, if any boundary.
+    All other may differ.
+INTEGRATION : integration by integration interfacing (one
+    #pulse_sequence of one exp_slice, then the next). exp_slice #1 and
+    exp_slice #2 must have same intt and intn. Integrations will
+    switch between one and the other until time is up/nave is
+    reached.
+PULSE : Simultaneous pulse_sequence interfacing, pulse by pulse
+    creates a single pulse_sequence. exp_slice A and B might have different
+    frequencies (stereo) and/or may have different pulse
+    length, mpinc, pulse_sequence, but must have the same integration
+    time. They must also have same len(scan), although they may
+    use different directions in scan. They must have the same
+    scan boundary if any. A time offset between the pulses
+    starting may be set (seq_timer in exp_slice). exp_slice A
+    and B will have integrations that run at the same time.
+"""
 
 
+debug_flag = True  # TODO move this somewhere central. config? - then all files could use it
```

Comment:
I think this would get incorporated into the scons stuff. I have flags in there for c++ that add different definitions for whether debug or release are getting built. Another option might be something like this https://stackoverflow.com/a/561338.